lib/single-new-eks-cost-monitoring-pattern/index.ts

```diff
@@ -0,0 +1,262 @@
+                        create: false,
+                        server: {
+                            create: false,
+                            name: "kubecost-prometheus-server-amp"
+                        }
+                    }
+                }                
+
+            }),
+            new blueprints.ArgoCDAddOn({
+                bootstrapRepo: {
+                    repoUrl: gitUrl,
+                    targetRevision: "main",
+                    path: 'secure-ingress-cognito/envs/dev'
+                },
+                bootstrapValues: {
+                    spec: {
+                        ingress: {
+                            host: subdomain,
+                            cognitoUserPoolArn: cognitoIdpStackOut.userPoolOut.userPoolArn,
+                            cognitoUserPoolAppId: cognitoIdpStackOut.userPoolClientOut.userPoolClientId,
+                            cognitoDomainName: cognitoIdpStackOut.userPoolDomainOut.domainName,
+                            certificateArn: certificate.certificateArn,
+                            region: process.env.CDK_DEFAULT_REGION,
+                        }
+                    },
+                },
+                adminPasswordSecretName: SECRET_ARGO_ADMIN_PWD,
+            }),
+            new KubecostServiceAccountsAddon()
+        ];
+
+        const mngProps: blueprints.MngClusterProviderProps = {
+            version: eks.KubernetesVersion.of("1.28"),
+            instanceTypes: [new ec2.InstanceType("m5.2xlarge")],
+            amiType: eks.NodegroupAmiType.AL2_X86_64,
+            desiredSize: 2,
+            maxSize: 3, 
+        };
+
+        await ObservabilityBuilder.builder()
+            .account(account)
+            .region(region)
+            .clusterProvider(new blueprints.MngClusterProvider(mngProps))
+            .version('auto')
+            .resourceProvider(GlobalResources.HostedZone, new LookupHostedZoneProvider(parentDomain))
+            .resourceProvider(GlobalResources.Certificate, new blueprints.CreateCertificateProvider('secure-ingress-cert', `${subdomain}`, GlobalResources.HostedZone))
+            .resourceProvider(ampWorkspaceName, new blueprints.CreateAmpProvider(ampWorkspaceName, ampWorkspaceName))
+            .addOns(...addOns)
+            .buildAsync(scope, stackId);
+    }
+}
+
+class KubeCostExtensionAddon extends KubecostAddOn {
+    constructor(props?: KubecostAddOnProps) {
+        super(props);
+    }
+
+    deploy(clusterInfo: blueprints.ClusterInfo): Promise<Construct> {
+        const region = process.env.COA_AWS_REGION! || process.env.CDK_DEFAULT_REGION!;
+        const ampWorkspaceId = this.options.values!.global.amp.prometheusServerEndpoint;
+        const prometheusServerEndpoint = 'http://localhost:8005/workspaces/' + ampWorkspaceId;
+        const remoteWriteEndpoint = `https://aps-workspaces.${region}.amazonaws.com/workspaces/${ampWorkspaceId}/api/v1/remote_write`;
```

Comment:
Is this accurate. You are passing `prometheusServerEndpoint` to an URL formation via `ampWorkspaceId`.

Reply:
Yes this is correct, please see here 
https://docs.kubecost.com/install-and-configure/advanced-configuration/custom-prom/amp-with-remote-write#preparing-the-configuration-file